selfdrive/ui/qt/onroad.cc

```diff
@@ -180,12 +180,13 @@ ExperimentalButton::ExperimentalButton(QWidget *parent) : QPushButton(parent) {
   setFixedSize(btn_size, btn_size);
   setCheckable(true);
 
-  params = Params();
   engage_img = loadPixmap("../assets/img_chffr_wheel.png", {img_size, img_size});
   experimental_img = loadPixmap("../assets/img_experimental.svg", {img_size, img_size});
 
-  QObject::connect(this, &QPushButton::toggled, [=](bool checked) {
-    params.putBool("ExperimentalMode", checked);
+  QObject::connect(this, &QPushButton::toggled, [this](bool checked) {
+    setEnabled(false);
+    Params().putBool("ExperimentalMode", checked);
+    QTimer::singleShot(1000, [this]() { setEnabled(true); });
```

Comment:
How about 100ms? I feel like the double clicks are tens of ms apart, not on the order of hundres

Reply:
yep, a second button press is registered about 20 ms after, I've reproduced this a few times now.

Reply:
> Though, it would be even better to debounce all UI buttons generically, can we do that?
y, think we can do it in another pr.

Reply:
there's a race condition with the onroad ExperimentalButton. the button signal `toggled` causes a param write, which controlsd reads and sends in `controlsState`, which `ExperimentalButton` reads to set checked...which writes the param again. So if you get a toggled signal, and then an old controlsState packet it will write twice!

you can  also write the param through another mean and ExperimentalButton will write it again (the `setChecked` call sends a `toggled` signal)

Reply:
Where is the parameter ExperimentalModeConfirmed set?

Reply:
controls.h `bool confirmed = store_confirm && params.getBool(key + "Confirmed");`